projects/js-packages/connection/components/use-connection/index.jsx

```diff
@@ -52,14 +52,15 @@ export default ( {
 	 * Initialize the site registration process.
 	 *
 	 * @param {Event} [e] - Event that dispatched handleRegisterSite
+	 * @returns {void}    - Promise when running the registration process. Otherwise, nothing.
 	 */
 	const handleRegisterSite = e => {
 		e && e.preventDefault();
 
 		if ( isRegistered ) {
 			handleConnectUser();
```

Comment:
Is it not a case to return this one too? To avoid returning `undefined` in some situations and `promises` in other

Reply:
A nice observation. I'm not sure, though.

The idea behind this PR change is to detect exactly when the site is registered. Here, at this point, the site is already registered so probably we wouldn't like to return a Promise.

Reply:
> The idea behind this PR change is to detect exactly when the site is registered.

Yep, got that point. But to keep consistency is not better to think that we're allowing the `handleRegisterSite` method to be detectable, which includes returning promise in all situations than only making `handleRegisterSite` detectable in 1 situation and not the other?

Reply:
Got your point! But I think we should refactor this part. 

Looking at the code, we're handling the `skip` of user connection wrong, we're doing that in `handleConnectUser` and it needs to be made here at `handleRegisterSite`.

This way we could return promises in all places and don't care about returning where they came from, because if we're using `skipUserConnection` it will always be related only with `registerSite` and if not it will be related with `connectUser`.

With that or mentality will be: Let's detect when `handleRegisterSite` finished and only that.

Showing that at code:

```es6
const handleConnectUser = () => connectUser( { from, redirectUri } );

const handleRegisterSite = e => {
  e && e.preventDefault();
  
  if ( isRegistered ) {
    return handleConnectUser();
  } else {
    return registerSite( { registrationNonce, redirectUri } ).then( () => {
      if ( ! skipUserConnection ) {
        return handleConnectUser();
      } else if ( redirectUri ) {
        window.location = redirectUri;
      }
    } );
  }
};
```

Reply:
> the problem here is handleRegisterSite() could run the resolve even when the site is already registered but indeed which we'd like to identify it

I think that's the point, as soon we move `skipUserConnection` to be handled inside `handleRegisterSite`, we know that if we're passing the `skipUserConnection` flag, the promise will resolve when site is registered, otherwise when user is connected, updating my last example:

```
const handleConnectUser = () => connectUser( { from, redirectUri } );

const handleRegisterSite = e => {
  e && e.preventDefault();

  if ( isRegistered && skipUserConnection ) {
    return Promise.resolve();
  }

  if ( isRegistered ) {
    return handleConnectUser();
  }
  
  return registerSite( { registrationNonce, redirectUri } ).then( () => {
    if ( ! skipUserConnection ) {
      return handleConnectUser();
    } else if ( redirectUri ) {
      window.location = redirectUri;
    }
  } );
};
```